public/app/core/components/TimelineChart/utils.ts

```diff
@@ -405,22 +400,13 @@ export function prepareTimelineFields(
               },
             },
           };
-          changed = true;
-          fields.push(field);
+          frames.push({
+            fields: [frame.fields[startFieldIdx], f],
+            length: frame.length,
+          });
           break;
         default:
-          changed = true;
-      }
-    }
-    if (isTimeseries && fields.length > 1) {
-      hasTimeseries = true;
-      if (changed) {
```

Comment:
I don't quite follow the purpose of this `changed` flag, so I might be missing something when refactoring this function to support "normalization".

Reply:
i _think_ it's an optimization so that we dont have to make frame/field copies, but will have to double-check.

Reply:
Thanks! I don't think this matters anymore since `prepareTimelineFields` is now untouched, so I'm resolving this conversation.